fiftyone/server/routes/embeddings.py

```diff
@@ -5,131 +5,297 @@
-        curr_sample_ids = results._curr_sample_ids
-        curr_label_ids = results._curr_label_ids
-        label_values = results.view.values(labels_field)
 
-        # TODO optimize count
+        # Color by data
+        label_values = results.view.values(labels_field, unwind=True)
         values_count = len(set(label_values))
-        selected_ids = get_selected_ids(
-            datasetName,
-            filters,
-            extended_stages,
-            extended_selection,
-            stages,
-            curr_sample_ids,
-        )
 
-        distinct_values = set(label_values)
+        # This is the total number of embeddings in `results`
+        index_size = results.total_index_size
+
+        # This is the number of embeddings in `results` that exist in `view`.
+        # Any operations that we do with `results` can only work with this data
+        available_count = results.index_size
+
+        # This is the number of samples/patches in `view` that `results`
+        # doesn't have embeddings for
+        missing_count = results.missing_size
+
+        curr_points = results._curr_points
+        if is_patches_plot:
+            curr_ids = results._curr_label_ids
+        else:
+            curr_ids = results._curr_sample_ids
+
+        #
+        # Now we need to decide whether to render any points as selected
+        #
+
+        if filters or extended_stages:
+            # Select points in extended view and deselect others
+            extended_view = fosv.get_view(
+                dataset_name,
+                stages=stages,
+                filters=filters,
+                extended_stages=extended_stages,
+            )
+            is_patches_view = extended_view._is_patches
+
+            if is_patches_plot and not is_patches_view:
+                _, id_path = extended_view._get_label_field_path(
+                    patches_field, "id"
+                )
+                extended_ids = extended_view.values(id_path, unwind=True)
+            elif is_patches_view and not is_patches_plot:
+                extended_ids = extended_view.values("sample_id")
+            else:
+                extended_ids = extended_view.values("id")
+
+            selected_ids = set(curr_ids) & set(extended_ids)
+        else:
+            # No filter is applied, everything can be selected
+            selected_ids = None
+
+        if extended_selection is not None:
+            if selected_ids:
+                selected_ids &= set(extended_selection)
+            else:
+                selected_ids = set(extended_selection)
+
+        if selected_ids is not None:
+            selected = [id in selected_ids for id in curr_ids]
+        else:
+            selected = itertools.repeat(True)
+
         style = (
-            "categorical"
-            if len(distinct_values) <= MAX_CATEGORIES
-            else "continuous"
+            "categorical" if values_count <= MAX_CATEGORIES else "continuous"
```

Comment:
I know you didn't change this, but this isn't exactly right. Any suggestions? Eg. if all the values are `1` it is still technically numeric.

I was going to try something like `get_field_type(labels_field) == Number`